substrate/frame/parameters/src/lib.rs

```diff
@@ -167,6 +168,22 @@ pub mod pallet {
 		type WeightInfo: WeightInfo;
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			let proof_size_before: u64 = get_proof_size().unwrap_or(0);
+
+			let items = Parameters::<T>::iter().count() as u64;
+
+			let proof_size_after: u64 = get_proof_size().unwrap_or(0);
+
+			let proof_size_diff = proof_size_after.saturating_sub(proof_size_before);
+
+			Weight::from_parts(0, proof_size_diff)
```

Comment:
Since `get_proof_size` depends on a host function that can be disabled, and the db cost computation is simple, we should probably use `T::DbWeight::get().reads(items)`
```suggestion
			T::DbWeight::get().reads(items)
```